### DISCUSSION TITLE
Add routine to import private key from OpenSSL EC private key format.

### DISCUSSION THREAD

**Discussion by unematiii** [panva/jose#317] [Q&A] [answered]
Is there plan to add support for importing OpenSSL EC private key format? Currently I only see an option for PKCS8.

**Comment by panva** [comment, marked as answer]
@unematiii could you be more specific about the key format? Why is PKCS#8 or JWK not suitable?

**Reply by unematiii** [reply]
TBH, simply lazyness. I decided to drop `auth0/node-jsonwebtoken` in favor to this library, as we need to deal with JWK(S). But since we have been historically using OpenSSL format for EC priv keys in certain situations, I need to explain inside team, as to why I am converting keys...

**Reply by panva** [reply]
So your runtime is node.js? You can use crypto.createPrivateKey with whatever PEM key format that worked for you before to obtain a KeyObject.

**Reply by panva** [reply]
I have no clue what you mean by Openssl Ec format btw... thats why i asked to be specific

**Reply by unematiii** [reply]
The private key is made available to me via env variable, and is generated using OpenSSL, st like:
`openssl ecparam -name prime256v1 -genkey -noout -out key.pem` (for legacy format [see](https://newbedev.com/difference-in-ecdsa-private-key-format)).

**Reply by panva** [reply]
That's a standard SEC1 format. Unfortunately it is not supported in web based runtimes. But as i pointed above you do not need to use the import routines at all, you can use node's crypto module instead.

**Reply by unematiii** [reply]
Thanks! We will use some conversion routines.

**Reply by panva** [reply]
But why? crypto.createPrivateKey(sec1string) works fine.

**Reply by unematiii** [reply]
Yes, thats what I meant to use.

**Comment by zxt-tzx** [comment]
Just wanna say this helped me! Thanks @panva for the really cool library! I managed to get things to work despite knowing next to nothing about JOSE :)
